markdownlint.sh

```diff
@@ -1,64 +1,113 @@
-          labels: $l,
-          close: {
-            stale: true,
-            message: "Closing pull request because all fixable warnings and/or errors have been fixed in base branch"
-          }
-        },
-        commit: {
-          message: "MarkdownLint fixes"
-        }
-    }'
-    )
-echo $push > /atm/output/push.json
+# print message to stderr prefixed by package name.
+# usage: err MESSAGE
+function err () {
+    msg "$*" 1>&2
+}
 
-##### Prepare command arguments
-if ! [[ -z "$fix" ]]; then
-  fix_option="--fix"
-fi
+function main () {
+    # Extract some skill configuration from the incoming event payload
+    local fix config ignores labels branch
+    fix=$( < "$ATOMIST_PAYLOAD" \
+               jq -r '.skill.configuration.instances[0].parameters[] | select( .name == "push_strategy" ) | .value' )
+    config=$( < "$ATOMIST_PAYLOAD" \
+                  jq -r '.skill.configuration.instances[0].parameters[] | select( .name == "config" ) | .value' )
+    ignores=$( < "$ATOMIST_PAYLOAD" \
+                   jq -r '.skill.configuration.instances[0].parameters[] | select( .name == "ignores" ) | .value | join(" --ignore ")' )
+    labels=$( < "$ATOMIST_PAYLOAD" \
+                  jq -r '.skill.configuration.instances[0].parameters[] | select( .name == "labels" ) | .value' )
+    branch=$( < "$ATOMIST_PAYLOAD" jq -r '.data.Push[0].branch' )
 
-if ! [[ -z "$config" ]] && ! [[ -f "/atm/home/.markdownlint.json" ]]; then
-  echo $config > "/atm/input/markdownlint.config.json"
-  config_option="--config /atm/input/markdownlint.config.json"
-fi
+    # Bail out early if it on a markdownlint branch
+    if [[ $branch == markdownlint-* ]]; then
+        exit 0
+    fi
 
-if ! [[ -z "ignores" ]]; then
-  ignore_option="--ignore $ignores"
-fi
+    local outdir=${ATOMIST_OUTPUT_DIR:-/atm/output}
 
-markdownlint "**/*.md" $config_option $ignore_option $fix_option
+    # Make the problem matcher available to the runtime
+    local matcher_outdir=$outdir/matchers
+    if ! mkdir -p "$matcher_outdir"; then
+        err "Failed to create matcher output directory: $matcher_outdir"
+        return 1
+    fi
+    if ! cp /app/markdownlint.matcher.json "$matcher_outdir"; then
+        err "Failed to copy markdownlink.matcher.json to $matcher_outdir"
+        return 1
+    fi
+
+    # Create push instructions for the runtime to indicate how changes to the repo should get persisted
+    if [[ ! $labels ]]; then
+        labels="[]"
+    fi
+
+    if ! > "$outdir/push.json" jq -n --arg s "$fix" --argjson l "$labels" '{
+    strategy: $s,
+    pullRequest: {
+      title: "MarkdownLint fixes",
+      body: "MarkdownLint fixed warnings and/or errors",
+      branchPrefix: "markdownlint",
+      labels: $l,
+      close: {
+        stale: true,
+        message: "Closing pull request because all fixable warnings and/or errors have been fixed in base branch"
+      }
+    },
+    commit: {
+      message: "MarkdownLint fixes"
+    }
+}'
```

Comment:
Like the problem extraction, tt is nice to have the power provided by the `push.json`, but I'd be happy with just a check based on the exit status.

Reply:
That’s already an option. You can get just a check.